src/azul/indexer.py

```diff
@@ -183,3 +183,72 @@ def merge(cur_bundles, new_bundles):
         for bundle in cur_bundles_by_id.values():
             assert bundles.setdefault(bundle['uuid'], bundle) is bundle
         return list(bundles.values())
+
+    def delete(self, dss_notification: Mapping[str, Any]) -> None:
+        bundle_uuid = dss_notification['match']['bundle_uuid']
+        bundle_version = dss_notification['match']['bundle_version']
+        index = dss_notification['match']['index']
+
+        es_client = self.properties.elastic_search_client
+
+        search_query = {
+            "query": {
+                "match": {
+                    "bundles.uuid.keyword": bundle_uuid
+                }
+            }
+        }
+
+        current_doc = es_client.search(index=index,
+                                       doc_type="doc",
+                                       body=search_query)
+
+        hits = current_doc['hits']['hits']
+
+        def update_call(cur_doc_id, body, version):
+            es_client.update(index=index,
+                             doc_type="doc",
+                             id=cur_doc_id,
+                             body=body,
+                             version=version
+                             )
+
+        def on_success():
+            response_to_return = es_client.search(index=index,
+                                                  doc_type="doc",
+                                                  body=search_query)
+
+            return response_to_return
+
+        def on_conflict(cur_doc_id, body, error):
+            log.warning("Deletion of doc %s failed with a conflict error %s . Retrying", current_doc_id, error)
+            hit_doc_updated = es_client.get(index=index,
+                                            doc_type="doc",
+                                            id=cur_doc_id)
+
+            update_call(current_doc_id, body, hit_doc_updated['_version'])
+
+        for hit in hits:
+            current_doc_id = hit['_id']
+            hit_doc = es_client.get(index=index,
+                                    doc_type="doc",
+                                    id=current_doc_id)
+
+            current_hit_doc = hit_doc['_source']
+            for bundle in current_hit_doc['bundles']:
+                if bundle['version'] == bundle_version:
+                    bundle['contents'] = None
+                    bundle['admin_deleted'] = True
+
+            update_doc = {
+                "doc": current_hit_doc
+            }
+            try:
+                doc_version = hit_doc['_version']
+                update_call(current_doc_id, update_doc, doc_version)
+            except ConflictError as e:
+                on_conflict(current_doc_id, update_doc, e)
+            except ElasticsearchException as e:
```

Comment:
Don't swallow exceptions like that unless you have a good reason to do so. Embellishing logs is not a good reason. You can log and reraise if you want to embellish the logs. But you should default to letting the natural exception handling runs its course. If you log an exception, use `log(…,exc_info=True)` or `log(…,exc_info=e)`. Prefer the former, catch without `as e` and use a bare `raise`. 

https://docs.python.org/3/library/exceptions.html